diff --git a/libvirt/tests/src/virsh_cmd/domain/virsh_vcpucount.py b/libvirt/tests/src/virsh_cmd/domain/virsh_vcpucount.py
index 324c4ad..3013f45 100644
--- a/libvirt/tests/src/virsh_cmd/domain/virsh_vcpucount.py
+++ b/libvirt/tests/src/virsh_cmd/domain/virsh_vcpucount.py
@@ -21,7 +21,8 @@ def reset_domain(vm, vm_state, needs_agent=False):
     vm_xml.set_vm_vcpus(vm.name, 4, 1)
     if not vm_state == "shut off":
         vm.start()
-        vm.prepare_guest_agent(prepare_xml=False)
+        if needs_agent:
+            vm.prepare_guest_agent(prepare_xml=False)
 
 
 def chk_output_running(output, expect_out, options):
diff --git a/libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py b/libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py
index 2f4bb83..75a126e 100644
--- a/libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py
+++ b/libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py
@@ -486,7 +486,7 @@ def run(test, params, env):
             if vm.is_alive():
                 vm.destroy(gracefully=False)
             xml_inst = vm_xml.VMXML.new_from_dumpxml(vm_name)
-            xml_inst.remove_agent_channel(vm_name)
+            xml_inst.remove_agent_channels()
             vm.start()
 
         # Record the previous snapshot-list
