diff --git a/libvirt/tests/cfg/virsh_cmd/domain/virsh_blockcopy.cfg b/libvirt/tests/cfg/virsh_cmd/domain/virsh_blockcopy.cfg                                               
index 91dcb26..6ccdd42 100644                                                                                                                                              
--- a/libvirt/tests/cfg/virsh_cmd/domain/virsh_blockcopy.cfg
+++ b/libvirt/tests/cfg/virsh_cmd/domain/virsh_blockcopy.cfg
@@ -1,6 +1,6 @@
 - virsh.blockcopy:
     type = virsh_blockcopy
-    target_disk = "vda"
+    target_disk = "sda"
     blockcopy_options = ""
     dest_path = ""
     dest_format = ""
diff --git a/libvirt/tests/cfg/virsh_cmd/snapshot/virsh_snapshot_create_as.cfg b/libvirt/tests/cfg/virsh_cmd/snapshot/virsh_snapshot_create_as.cfg
index 51df3d9..f02fd76 100644
--- a/libvirt/tests/cfg/virsh_cmd/snapshot/virsh_snapshot_create_as.cfg
+++ b/libvirt/tests/cfg/virsh_cmd/snapshot/virsh_snapshot_create_as.cfg
@@ -57,7 +57,7 @@
                     snap_createas_opts = "a--a a--a--snap1"
                 - disk_only_spec:
                     snap_createas_opts = "--disk-only"
-                    diskspec_opts = "vda,snapshot=external,driver=qcow2,file=disk-snap.img"
+                    diskspec_opts = "sda,snapshot=external,driver=qcow2,file=disk-snap.img"
                 - check_libvirtd_log:
                     check_json_no_savevm = "yes"
                     snap_createas_opts = "--disk-only"
